modules/auth_plugins/lib/open_project/plugins/auth_plugin.rb

```diff
@@ -47,11 +47,15 @@ def self.strategies
     end
 
     def self.providers_for(strategy)
-      strategies[strategy_key(strategy)].map(&:call).flatten.map(&:to_hash)
+      filtered_strategies strategies[strategy_key(strategy)].map(&:call).flatten.map(&:to_hash)
     end
 
     def self.providers
-      strategies.values.flatten.map(&:call).flatten.map(&:to_hash)
+      filtered_strategies strategies.values.flatten.map(&:call).flatten.map(&:to_hash)
+    end
+
+    def self.filtered_strategies(options)
+      options.select { |provider| !::EnterpriseToken.show_banners? || provider[:name]&.to_s == 'developer' }
```

Comment:
It would be nice if users could find out that this is happening. I.e. if we logged that strategy so-and-so is ignored because one needs an Enterprise edition for that or something along those lines.